app/forms/concerns/wizard.rb

```diff
@@ -0,0 +1,61 @@
+module Wizard
+  extend ActiveSupport::Concern
+
+  included do
+    attr_writer :state_store
+    attr_accessor :current_step, :review_mode
+  end
+
+  def initialize(state_store, params = {})
+    @state_store = state_store
+    sanitize_attributes(params)
+
+    super(last_saved_state.deep_merge(params))
+
+    update_review_mode if current_step.eql?('check')
+  end
+
+  def next_step
+    :check
+  end
+
+  def previous_step
+    :new
+  end
+
+  def clear_state!
+    @state_store.delete
+  end
+
+  alias_method :save, :clear_state!
+
+  def save_state!
+    @state_store.write(state)
+  end
+
+  def valid?
+    current_step.present? ? super(current_step.to_sym) : super
+  end
+
+private
+
+  def state
+    as_json(except: ignored_params).to_json
+  end
+
+  def last_saved_state
+    saved_state = @state_store.read
+    saved_state ? JSON.parse(saved_state) : {}
+  end
+
+  def ignored_params
```

Comment:
Could this also have a clearer name — as it's a form it almost implies that we're rejecting params coming from the front end? How about `params_to_exclude_from_stored_state`?